### PR TITLE
Namespace Creation errors

### PR DESCRIPTION
**Pull request by deepakwadhwani**
Fixes for : 
https://issues.cask.co/browse/CDAP-8929
https://issues.cask.co/browse/CDAP-8928
Improve error reporting logs.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DistributedStorageProviderNamespaceAdmin.java [hunk 1]
```diff
@@ -106,8 +106,9 @@ public void create(NamespaceMeta namespaceMeta) throws IOException, ExploreExcep
 
     try (HBaseAdmin admin = new HBaseAdmin(hConf)) {
       if (!tableUtil.hasNamespace(admin, hbaseNamespace)) {
-        throw new IOException(String.format("Custom mapped HBase namespace doesn't exist %s for namespace %s",
-                                            hbaseNamespace, namespaceMeta.getName()));
+        throw new IOException(String.format("HBase namespace '%s' specified for custom mapped namespace '%s' does not" +
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DistributedStorageProviderNamespaceAdmin.java]: can we remove the words "custom mapped"? Its a very developer-ish term, that users may not understand. It doesn't even matter in this context right? Removing it completely should also work.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java [hunk 2]
```diff
@@ -138,7 +138,8 @@ private void createLocation(NamespaceMeta namespaceMeta) throws IOException {
       // no namespace custom location was provided one must be created by cdap
       namespaceHome = namespacedLocationFactory.get(namespaceMeta);
       if (namespaceHome.exists()) {
-        throw new FileAlreadyExistsException(namespaceHome.toString());
+        throw new FileAlreadyExistsException(null, null,
+                                             String.format("Namespace home path %s already exists.", namespaceHome));
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java]: HDFS directory <> for namespace <> already exists?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DistributedStorageProviderNamespaceAdmin.java (hunk 1 above)
Comment (deepakwadhwani) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DistributedStorageProviderNamespaceAdmin.java]: Sure, I can remove it, however documentation for custom mapped namespaces does exist http://docs.cask.co/cdap/4.1.1/en/developers-manual/building-blocks/namespaces.html#custom-mapping-of-storage-providers

**Comment by deepakwadhwani** [comment]
@bdmogal please check again.

**Review by bdmogal — APPROVED** [review]

**Comment by deepakwadhwani** [comment]
clean build: https://builds.cask.co/browse/CDAP-RUT991-1

**Pull request merged**